x-pack/plugins/reporting/server/config/schema.ts

```diff
@@ -101,6 +101,45 @@ const PollSchema = schema.object({
   }),
 });
 
+const ExportTypeSchema = schema.object({
+  // Csv reports are enabled in all offerings
+  csv: schema.object({ enabled: schema.boolean({ defaultValue: true }) }),
+  // Png reports are disabled in serverless
+  png: schema.object({
+    enabled: schema.conditional(
+      schema.contextRef('serverless'),
+      true,
+      schema.boolean({
+        validate: (rawValue) => {
+          if (rawValue === true) {
+            return 'PNG reports can only be disabled in serverless mode';
+          }
+        },
+        // when running in dev mode default to true
+        defaultValue: schema.contextRef('dev'),
+      }),
+      schema.boolean({ defaultValue: false })
```

Comment:
1. I think this code is not setting the right defaults, and the `validate` method doesn't really seem to function the way the message says.
2. And it's not very important to enable PNG/PDF for serverless in dev mode.
3. The final defaultValue outside of serverless should be `true` so the current behavior isn't getting changed.

For simplicity, let's not worry if on-prem users find this setting and decide they want to disable PNG/PDF reports.  That allows us to not have the `validate` methods: we just need to set the right defaults in the config here.

The thing I suggest to make it simple is: don't use the `serverless` contextRef in the schema. 

How about:
```typescript
const ExportTypeSchema = schema.object({
  // Csv reports are enabled in all offerings
  csv: schema.object({
    enabled: schema.boolean({ defaultValue: true }),
  }),
  // Png reports are disabled in serverless
  png: schema.object({
    enabled: schema.conditional(
      schema.contextRef('serverless'),
      true,
      schema.boolean({ defaultValue: false }),
      schema.boolean({ defaultValue: true })
    ),
  }),
  // Pdf reports are disabled in serverelastless
  pdf: schema.object({
    enabled: schema.conditional(
      schema.contextRef('serverless'),
      true,
      schema.boolean({ defaultValue: false }),
      schema.boolean({ defaultValue: true })
    ),
  }),
});
```

Reply:
d76850e for changes. I removed some of the tests because those will fail and aren't relevant if we aren't going to worry about enablement of image reporting in serverless by a user (mocking that someone tries to enable pdf or png with the serverless config)